android/gltfio-android/src/main/java/com/google/android/filament/gltfio/MaterialProvider.java

```diff
@@ -99,9 +99,10 @@ public void constrainMaterial(@NonNull @Size(min = 8) int[] uvmap) {
      * @param uvmap Output argument that gets populated with a small table that maps from a glTF uv
      *              index to a Filament uv index (0 = UNUSED, 1 = UV0, 2 = UV1).
      * @param label Optional tag that is not a part of the cache key.
+     * @param extras Optional extras that are not a part of the cache key.
      */
     public @Nullable MaterialInstance createMaterialInstance(MaterialKey config,
-            @NonNull @Size(min = 8) int[] uvmap, @Nullable String label);
+            @NonNull @Size(min = 8) int[] uvmap, @Nullable String label, @Nullable String extras);
```

Comment:
This is unnecessary API breakage. You should keep the existing method, create a new one with the new argument, and have the old method call the new one by passing `null` for the extras.

Reply:
May I ask who actually uses that API except for Filament itself?

Reply:
You apparently for starters :) And it doesn't matter, it's a public API and should be treated as such. We sometimes break them when we can't do otherwise but this is very easy to support properly.

Reply:
BTW you should also document the format of that string. What does it contain? JSON?

Reply:
> You apparently for starters :) 

Kind-a yes, since extras support was our initiative.

> And it doesn't matter, it's a public API and should be treated as such. We sometimes break them when we can't do otherwise but this is very easy to support properly.

The API was introduced a month ago, and Filament breaks API every now and then. In patch releases.

> engine: Removed the "EVILS" tone mapping operator [⚠️ API Change].

And at the same time requesting to be strict in backward compatibility?

> BTW you should also document the format of that string. What does it contain? JSON?

Yes, JSON or null. Valid point.

Reply:
> And at the same time requesting to be strict in backward compatibility?

Yes we do break backward compatibility sometimes, that doesn't mean we do it *every* time. For instance we recently changed how the tone mappers work, but we kept the older API and re-implemented it on top of the new API for compatibility. This is a very easy change to make here so there's no reason not to do it.

Reply:
I should note that there's no base class for `MaterialProvider`, so the interface would have to have two methods, and it will be a breaking change in any case since all derived classes would have to implement the extra method.
I'm not sure I follow which would be a better way.

Reply:
Default methods could work here.

Reply:
That said, if this is only called by the JNI code then I guess we can break the API here. It would otherwise be complicated to figure out which version to invoke.

Reply:
If I got that right, the sole purpose of MaterialProvider existence is to be used from JNI side.

Reply:
Yes this is actually an "internal API" whose raison d'être is our JNI layer, so it's probably okay to break it as long we note it in the release notes.

I know it's been a loooong time  (my apologies) but I actually think we can merge this PR if it doesn't break anything.  I'll go ahead and do so.